charts/gsp-cluster/templates/02-gsp-system/service-operator/deployment.yaml

```diff
@@ -18,6 +18,7 @@ spec:
   template:
     metadata:
       annotations:
+        prometheus.io/scrape: "true"
```

Comment:
I don't think this will do anything - we have a `kubernetes-pods-istio-secure` scrape_config which looks for this annotation, but only if it also has a `istio-mtls: true` annotation which nothing currently has.

Options:

 - we build a ServiceMonitor for this specific service
 - we fix the prometheus config to make this annotation work the way we think it should

either way, it shouldn't block this PR (it's big enough already) but we should note it and fix it

Reply:
You are correct. This is something that has been generated with `kubebuilder` and essentially will not work without `ServiceMonitor` - which is my preferred option.

This will not be regenerated, therefore I've left it in half finished, but `ServiceMonitor` is not in scope. Hopefully, that means there will be a small story required to get the metrics later on.

Reply:
_wontfix_